falcon/request.py

```diff
@@ -1495,3 +1697,36 @@ def __init__(self):
         self.strip_url_path_trailing_slash = True
         self.default_media_type = DEFAULT_MEDIA_TYPE
         self.media_handlers = Handlers()
+
+
+class Forwarded(object):
+    """Represents a parsed Forwarded header.
+
+    (See also: RFC 7239, Section 4)
+
+    Attributes:
+        by (str): The value of the "by" parameter, or
+            ``None`` if the parameter was absent. Identifies the
+            client-facing interface of the proxy.
+        client (str): The value of the "for" parameter, or
+            ``None`` if the parameter was absent. Identifies the
+            node making the request to the proxy.
+        host (str): The value of the "host" parameter, or
+            ``None`` if the parameter was absent. Provides the host
+            request header field as received by the proxy.
+        scheme (str): The value of the "proto" parameter, or
+            ``None`` if the parameter was absent. Indicates what
+            protocol was originally used to make the request to the
+            proxy.
+    """
+
+    # NOTE(kgriffs): Use "client" since "for" is a keyword, and
+    # "scheme" instead of "proto" to be consistent with the
+    # falcon.Request interface.
+    __slots__ = ('by', 'client', 'host', 'scheme')
+
+    def __init__(self):
+        self.by = None
+        self.client = None
```

Comment:
TBH, I think target or destination more accurately describe the "by" parameter. According to RFC 7239:

> The "for" parameter is used to disclose information about the client that initiated the request and subsequent proxies in a chain of proxies.

Whereas:

>  The "by" parameter is used to disclose the interface where the request came in to the proxy server.

What do you think about the following variations?

```py
forwarded = req.forwarded[0]

forwarded.source
forwarded.dest

forwarded.source
forwarded.dest

forwarded.client
forwarded.target

forwarded.client
forwarded.dest

forwarded.client
forwarded.proxy

forwarded.client
forwarded.by
```